x-pack/plugins/infra/public/alerting/log_threshold/components/alert_details_app_section/components/logs_history_chart.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import moment from 'moment';
+import React from 'react';
+import { Rule } from '@kbn/alerting-plugin/common';
+import { EuiPanel, EuiFlexGroup, EuiFlexItem, EuiTitle, EuiText, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { convertTo, TopAlert } from '@kbn/observability-plugin/public';
+import { AnnotationDomainType, LineAnnotation, Position } from '@elastic/charts';
+import { EuiIcon, EuiBadge } from '@elastic/eui';
+import { useFetchTriggeredAlertsHistory } from '../../../../../hooks/use_fetch_triggered_alert_history';
+import { type PartialCriterion } from '../../../../../../common/alerting/logs/log_threshold';
+import { CriterionPreview } from '../../expression_editor/criterion_preview_chart';
+import { PartialRuleParams } from '../../../../../../common/alerting/logs/log_threshold';
+
+const LogsHistoryChart = ({
+  rule,
+  alert,
+}: {
+  rule: Rule<PartialRuleParams>;
+  alert: TopAlert<Record<string, any>>;
+}) => {
+  // Show the Logs History Chart ONLY if we have one criteria
+  // So always pull the first criteria
+  const criteria = rule.params.criteria[0];
+  const dateNow = Date.now();
+  const dateLast30Days = Number(moment(dateNow).subtract(30, 'days').format('x'));
```

Comment:
**suggestion:** the date range defined here is not synced with the time range defined inside the `useFetchTriggeredAlertsHistory` definition.
If we decide to change the time range to a different one in future, we'll need to update this here and also inside the hook definition.

To have a more flexible API, you can make the `useFetchTriggeredAlertsHistory`custom hook accept also a custom date-range, so that you can define it here as a shared source of truth for both the custom hook and `CriterionPreview` component. It could end up being something like:
```js
const dateRange = {
  from: 'now-30d',
  to: 'now'
}

const executionTimeRange = {
  gte: DateMath.parse(dateRange.from).valueOf()
  lte: DateMath.parse(dateRange.to, { roundUp: true }).valueOf()
}

useFetchTriggeredAlertsHistory({
  dateRange,
  // etc...
})

<CriterionPreview executionTimeRange={executionTimeRange} />
```

Reply:
Good idea! applied